server/apps/microservices/src/processors/video-transcode.processor.ts

```diff
@@ -0,0 +1,57 @@
+import { Process, Processor } from '@nestjs/bull';
+import { Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Job } from 'bull';
+import ffmpeg from 'fluent-ffmpeg';
+import { existsSync, mkdirSync } from 'fs';
+import { Repository } from 'typeorm';
+import { AssetEntity } from '../../../../libs/database/src/entities/asset.entity';
+import { APP_UPLOAD_LOCATION } from '../../../immich/src/constants/upload_location.constant';
+
+@Processor('video-conversion-queue')
+export class VideoTranscodeProcessor {
+  constructor(
+    @InjectRepository(AssetEntity)
+    private assetRepository: Repository<AssetEntity>,
+  ) {}
+
+  @Process({ name: 'mp4-conversion', concurrency: 1 })
```

Comment:
The `concurrency: 1` option limits one worker in the pool to handle this process, any additional videos that are added to the queue will be in the waiting queue until the worker is idle.